File: c/raindrops/raindrops.c
#include "raindrops.h"

#include <stdio.h>
#include <string.h>

// Raindrops is a twist on the FizzBuzz challenge.
void convert(char result[], int drops)
{
    if (drops % 3 == 0) {
        strcat(result, "Pling");
    }
    if (drops % 5 == 0) {
        strcat(result, "Plang");
    }
    if (drops % 7 == 0) {
        strcat(result, "Plong");
    }

    if (!strlen(result)) {
        sprintf(result, "%d", drops);
    }
}

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

func Convert(number int) string {
    sounds := ""

	if number % 3 == 0 {
        sounds += "Pling"
    }

	if number % 5 == 0 {
        sounds += "Plang"
    }

	if number % 7 == 0 {
        sounds += "Plong"
    }

    if len(sounds) == 0 {
        sounds = strconv.Itoa(number)
    }

    return sounds
}
